app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileEditDialog.java

```diff
@@ -93,11 +93,14 @@ boolean isEmailValid(CharSequence email) {
     /// check if entered phone number is valid or not
     // if not, show message
     boolean isPhoneValid(String phone) {
+        if (phone.length() == 0){
+            return true;
+        }
         boolean valid =  android.util.Patterns.PHONE.matcher(phone).matches();
         if (!valid){
             editPhone.setError(getString(R.string.m_not_valid_phone));
```

Comment:
i think you missed a `return false` after line 101. Please add that and then PR looks good to merge.